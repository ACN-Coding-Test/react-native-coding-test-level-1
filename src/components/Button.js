import  React from 'react';
import PropTypes from 'prop-types'
import { TouchableOpacity,StyleSheet,Text } from 'react-native';

import theme from "../theme";

function Button(props) {
  const {onPress,title}=props; 
  return (
  <TouchableOpacity onPress={onPress} style={styles.buttonContainer}>
      <Text style={styles.titleStyle}>{title}</Text>
  </TouchableOpacity>
  );
}

Button.defaultProps = {
    title:"",
    onPress:()=>{}
 }

Button.propTypes = {
   title:PropTypes.string,
   onPress:PropTypes.func
}
  

const styles = StyleSheet.create({
    buttonContainer: {
      backgroundColor:theme.colors.primary,
      paddingHorizontal:theme.padding.default,
      paddingVertical:theme.padding.default,
      borderRadius:theme.borderRadius.default,
      minWidth:150,
      alignItems:"center"
    },
    titleStyle:{
        color:theme.colors.textColor,
        fontSize:theme.fontSizes.body
    }
});


export default Button;
