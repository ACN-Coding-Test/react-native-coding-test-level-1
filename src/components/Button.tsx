import React from 'react';
import {TextInput, TouchableOpacity, View} from 'react-native';
import {Subheading, Surface} from 'react-native-paper';

type Props = {
  label: string;
  onPress: () => void;
  backgroundColor: string;
};

const Button = ({label, onPress, backgroundColor}: Props) => {
  return (
    <Surface
      style={{
        width: '90%',
        elevation: 5,
        height: 60,
        borderRadius: 5,
        marginTop: '5%',
      }}>
      <TouchableOpacity
        onPress={onPress}
        style={{
          flex: 1,
          backgroundColor: backgroundColor,
          borderRadius: 5,
          alignItems: 'center',
          justifyContent: 'center',
        }}>
        <Subheading style={{color: 'white', letterSpacing: 1.5, fontSize: 18}}>
          {label}
        </Subheading>
      </TouchableOpacity>
    </Surface>
  );
};

export default Button;
