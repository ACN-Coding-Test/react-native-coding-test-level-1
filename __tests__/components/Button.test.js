import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../../src/components/Button'
import { render, fireEvent } from '@testing-library/react-native'

describe('Button:', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should onPress fire correctly', () => {
    const onEventMock = jest.fn()
    const { getByText } = render(
      <Button title={'Button'} onPress={onEventMock} />,
    )
    fireEvent.press(getByText('Button'))
    expect(onEventMock).toHaveBeenCalled()
  })

})

