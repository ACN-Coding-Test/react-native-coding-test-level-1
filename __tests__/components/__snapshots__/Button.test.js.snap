// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button: should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#A100FF",
      "borderRadius": 8,
      "minWidth": 150,
      "opacity": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
      }
    }
  >
    
  </Text>
</View>
`;
